stl/inc/flat_map

```diff
@@ -960,210 +886,530 @@ private:
+    }
+
+    template <class _MappedTy>
+    iterator insert_or_assign(const_iterator _Position, const key_type& _Key_val, _MappedTy&& _Mapped_val)
+        requires is_assignable_v<mapped_type&, _MappedTy> && is_constructible_v<mapped_type, _MappedTy>
+    {
+        return this->_Emplace_hint<true>(_Position, _Key_val, _STD forward<_MappedTy>(_Mapped_val));
+    }
+
+    template <class _MappedTy>
+    iterator insert_or_assign(const_iterator _Position, key_type&& _Key_val, _MappedTy&& _Mapped_val)
+        requires is_assignable_v<mapped_type&, _MappedTy> && is_constructible_v<mapped_type, _MappedTy>
+    {
+        return this->_Emplace_hint<true>(_Position, _STD move(_Key_val), _STD forward<_MappedTy>(_Mapped_val));
+    }
+
+    template <class _OtherKey, class _MappedTy>
+    iterator insert_or_assign(const_iterator _Position, _OtherKey&& _Key_val, _MappedTy&& _Mapped_val)
+        requires _Is_transparent_v<key_compare> && is_constructible_v<key_type, _OtherKey>
+              && is_assignable_v<mapped_type&, _MappedTy> && is_constructible_v<mapped_type, _MappedTy>
+    {
+        return this->_Emplace_hint<true>(
+            _Position, _STD forward<_OtherKey>(_Key_val), _STD forward<_MappedTy>(_Mapped_val));
+    }
+
 private:
-    using _MyBase = _Flat_Map_Base<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer, true,
-        flat_multimap<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer>>;
+    using _MyBase::_Erase_if;
 
-public:
-    using _MyBase::_MyBase;
-    using _MyBase::operator=;
+    template <class _KTy, class _MTy, class _Comp, class _KeyCont, class _MappedCont, class _Pred>
+    friend typename flat_map<_KTy, _MTy, _Comp, _KeyCont, _MappedCont>::size_type erase_if(
+        flat_map<_KTy, _MTy, _Comp, _KeyCont, _MappedCont>&, _Pred);
+
+    template <class _KeyTy>
+    _NODISCARD mapped_type& _At(const _KeyTy& _Key_val)
+        requires same_as<_KeyTy, key_type> || _Is_transparent_v<key_compare>
+    {
+        const auto _Position = this->find(_Key_val);
+        if (_Position == this->end()) {
+            _Xout_of_range("std::flat_map::at: the specified key does not exist.");
+        } else {
+            return _Position->second;
+        }
+    }
+
+    template <class _KeyTy>
+    _NODISCARD const mapped_type& _At(const _KeyTy& _Key_val) const
+        requires same_as<_KeyTy, key_type> || _Is_transparent_v<key_compare>
+    {
+        const auto _Position = this->find(_Key_val);
+        if (_Position == this->end()) {
+            _Xout_of_range("std::flat_map::at: the specified key does not exist.");
+        } else {
+            return _Position->second;
+        }
+    }
+
+    template <class _KeyTy, class... _MappedArgTypes>
+    pair<iterator, bool> _Try_emplace(_KeyTy&& _Key_val, _MappedArgTypes&&... _Mapped_args) {
+        auto _Key_it = _STD lower_bound(_Data.keys.begin(), _Data.keys.end(), _Key_val, _Key_compare);

```

Comment:
Pre-existing (and massive in scope): This container isn't performing unwrapping when it knows it's giving correct iterator pairs to algorithms. Not a huge deal (the impact is debug mode only, and the algorithms themselves will check once and then unwrap).